tests/handlers/records-subscribe.spec.ts

```diff
@@ -0,0 +1,773 @@
+          expect(subscriptionReply.status.code).to.equal(200);
+          expect(subscriptionReply.subscription).to.exist;
+
+
+          // Alice writes a 'friend' $globalRole record with Bob as recipient
+          const friendRoleRecord = await TestDataGenerator.generateRecordsWrite({
+            author       : alice,
+            recipient    : bob.did,
+            protocol     : protocolDefinition.protocol,
+            protocolPath : 'friend',
+            data         : new TextEncoder().encode('Bob is my friend'),
+          });
+          const friendRoleReply = await dwn.processMessage(alice.did, friendRoleRecord.message, { dataStream: friendRoleRecord.dataStream });
+          expect(friendRoleReply.status.code).to.equal(202);
+
+          const recordIds: string[] = [];
+          const addRecord = async (message: GenericMessage): Promise<void> => {
+            if (message.descriptor.interface === DwnInterfaceName.Records && message.descriptor.method === DwnMethodName.Write) {
+              const recordsWriteMessage = message as RecordsWriteMessage;
+              recordIds.push(recordsWriteMessage.recordId);
+            }
+          };
+
+          // subscribe with friend role
+          const bobSubscriptionWithRole = await TestDataGenerator.generateRecordsSubscribe({
+            filter,
+            author       : bob,
+            protocolRole : 'friend',
+          });
+
+          const subscriptionWithRoleReply = await dwn.processMessage(alice.did, bobSubscriptionWithRole.message, { subscriptionHandler: addRecord });
+          expect(subscriptionWithRoleReply.status.code).to.equal(200);
+          expect(subscriptionWithRoleReply.subscription).to.exist;
+
+
+          // Alice writes three 'chat' records
+          const chatRecordIds = [];
+          for (let i = 0; i < 3; i++) {
+            const chatRecord = await TestDataGenerator.generateRecordsWrite({
+              author       : alice,
+              recipient    : alice.did,
+              protocol     : protocolDefinition.protocol,
+              protocolPath : 'chat',
+              published    : false,
+              data         : new TextEncoder().encode('Bob can read this cuz he is my friend'),
+            });
+            const chatReply = await dwn.processMessage(alice.did, chatRecord.message, { dataStream: chatRecord.dataStream });
+            expect(chatReply.status.code).to.equal(202);
+            chatRecordIds.push(chatRecord.message.recordId);
+          }
+
+          // there should not be any messages in the subscription without a friend role.
+          expect(noRoleRecords.length).to.equal(0);
+
+          // should have all chat messages
+          expect(recordIds).to.have.members(chatRecordIds);
+        });
+
+        it('allows protocol authorized subscriptions', async () => {
+          // scenario: Alice writes some chat messages writes a chat message.
```

Comment:
Typo in the original `RecordsQuery` test.

```suggestion
          // scenario: Alice writes some chat messages.
```

Reply:
Fixed in both!